### PR TITLE
Splitting submodules of project

### PR DESCRIPTION
**Pull request by winterheart**
## Pull Request Type

- [ ] GitHub Workflow changes
- [ ] Documentation or Wiki changes
- [x] Build and Dependency changes
- [ ] Runtime changes
  - [ ] Render changes
  - [ ] Audio changes
  - [ ] Input changes
  - [ ] Network changes
  - [ ] Other changes

### Description

Isolate and cleanup some submodules of project:
* 2dlib
* AudioEncode
* d3music
* ui
* grtext
* model
* physics
* rtperformance
* unzip
* libacm

Remove unused submodules:
* czip

### Checklist

- [x] I have tested my changes locally and verified that they work as intended.
- [x] I have documented any new or modified functionality.
- [x] I have reviewed the changes to ensure they do not introduce any unnecessary complexity or duplicate code.
- [x] I understand that by submitting this pull request, I am agreeing to license my contributions under the project's license.

### CODE REVIEW

File: unzip/CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,10 @@
-set(HEADERS)
 set(CPPS unzip.cpp)
 
 add_library(unzip STATIC ${HEADERS} ${CPPS})
 
-target_link_libraries(unzip ZLIB::ZLIB)
+target_link_libraries(unzip PUBLIC ZLIB::ZLIB)
```
Comment (Lgt2x) [inline unzip/CMakeLists.txt:5]: this should be `PRIVATE`, as consumers should not use the `ZLIB::ZLIB` target directly

File: AudioEncode/CMakeLists.txt [hunk 2]
```diff
@@ -1,6 +1,13 @@
-set(HEADERS)
```
Comment (Lgt2x) [inline AudioEncode/CMakeLists.txt:1]: I'd strongly advise for listing headers in the sources for every module. As we've seen with #411 , this improves IDE integration among other things. It's fine you don't want to deal with that in this PR, I can also take care of it afterwards

File: AudioEncode/encoder.cpp [hunk 3]
```diff
@@ -90,7 +90,7 @@ bool aenc_Compress(char *input_filename, char *output_filename, int *input_level
     factor = *input_factor; // Factor of compression (default 4 for 22K, 8 for 44K)
     factor_set = 1;
 
-    if (factor < 1.0f && factor)
+    if (factor > 0.0f && factor < 1.0f)
```
Comment (Lgt2x) [inline AudioEncode/encoder.cpp:93]: this changes logic, is that intended? We don't invert negative factors now

File: grtext/grfont.cpp [hunk 4]
```diff
@@ -655,7 +654,7 @@ bool grfont_SetTemplate(const char *pathname, const tFontTemplate *ft) {
 
   //	Write widths now if necessary.(FT_PROPORTIONAL)
   if (fnt.flags & FT_PROPORTIONAL) {
-    for (int i = 0; i < num_char; i++)
+    for (i = 0; i < num_char; i++)
```
Comment (Lgt2x) [inline grtext/grfont.cpp:657]: where is that `i` declared? It would be better to keep it a loop variable

File: libacm/CMakeLists.txt [hunk 5]
```diff
@@ -1,11 +1,17 @@
...
+  adecode.cpp
   aencode.cpp
-  adecode.cpp)
+)
 
 # these are the relevant source files from upstream libacm (https://github.com/markokr/libacm/)
 set(LIB_SRC
   decode.c
-  libacm.h)
+  libacm.h
```
Comment (Lgt2x) [inline libacm/CMakeLists.txt:9]: move into `third_party`?

File: model/polymodel.cpp [hunk 6]
```diff
@@ -1816,23 +1807,23 @@ int ReadNewModelFile(int polynum, CFILE *infile) {
...
 
-        ASSERT(pm->submodel[i].keyframe_axis != NULL);
-        ASSERT(pm->submodel[i].keyframe_angles != NULL);
-        ASSERT(pm->submodel[i].keyframe_matrix != NULL);
+        ASSERT(pm->submodel[i].keyframe_axis != nullptr);
+        ASSERT(pm->submodel[i].keyframe_angles != nullptr);
+        ASSERT(pm->submodel[i].keyframe_matrix != nullptr);
 
-        for (int t = 0; t < pm->submodel[i].num_key_angles; t++) {
+        for (t = 0; t < pm->submodel[i].num_key_angles; t++) {
```
Comment (Lgt2x) [inline model/polymodel.cpp:1826]: I'm really not fond of loop variables declared outside of said loop

File: physics/findintersection.cpp [hunk 7]
```diff
@@ -855,37 +855,27 @@
...
-#include "polymodel.h"
-#include "PHYSICS.H"
-#include "player.h"
-#include "doorway.h"
-#include "renderer.h"
-
-// Debug performance includes (do nothing in final release)
-#ifndef NED_PHYSICS
-#include "rtperformance.h"
+#include "../neweditor/ned_GameTexture.h"
```
Comment (Lgt2x) [inline physics/findintersection.cpp:878]: this header does not exist, remove

File: physics/physics.cpp [hunk 8]
```diff
@@ -554,33 +536,20 @@ bool PhysicsDoSimRot(object *obj, float frame_time, matrix *orient, vector *rott
...
-            //						mprintf(0, "4 scale %f\n", scale);
           }
         }
       }
     }
 
     if (!f_pitch_leveled && fabs(rotthrust->z) < 100.0f) {
-      if (1) ///*&& (pi->turnroll <= 10 || pi->turnroll >= 65535 - 10)*/)
-      {
+      if (true) {
```
Comment (Lgt2x) [inline physics/physics.cpp:552]: remove condition altogether

File: AudioEncode/encoder.cpp (hunk 3 above)
Comment (winterheart) [reply AudioEncode/encoder.cpp]: There was logic checking that factor is non-zero. Checking for negative value is next after this. Still, I'll change it back to `factor != 0.0f`.

File: libacm/CMakeLists.txt (hunk 5 above)
Comment (winterheart) [reply libacm/CMakeLists.txt]: Splatted into `third_party/libacm` and `AudioEncode`.

File: grtext/grfont.cpp (hunk 4 above)
Comment (winterheart) [reply grtext/grfont.cpp:657]: There `i` usage out of scope in `grfont_SetTemplate()` after loop. This is only cheap and quick way to silence "overshadowed local variable" warning.

File: model/polymodel.cpp (hunk 6 above)
Comment (winterheart) [reply model/polymodel.cpp:1826]: Same as earlear - out of loop usage in function.

File: AudioEncode/CMakeLists.txt (hunk 2 above)
Comment (winterheart) [reply AudioEncode/CMakeLists.txt:1]: Let's make it in another PR.

**Review by Lgt2x — APPROVED** [review]
this one is good to go

**Pull request merged**